Cython/Compiler/Code.py

```diff
@@ -1619,27 +1629,21 @@ def generate_string_constants(self):
                 else:
                     encoding = '"%s"' % py_string.encoding.lower()
 
-                self.parts['module_state'].putln("PyObject *%s;" % py_string.cname)
-                self.parts['module_state_defines'].putln("#define %s %s->%s" % (
+                self.parts['module_state_defines'].putln("#define %s %s->%s[%s]" % (
```

Comment:
When we get do the module state refactoring to get rid of most of these defines, I'd propose keeping this as

```
#define %s %s[%s]
```

so we at least have a mapping from "informative cname" to "array lookup"